external/model-preparation-algorithm/mpa_tasks/apis/classification/task.py

```diff
@@ -148,11 +156,18 @@ def _init_recipe(self):
 
         self._recipe_cfg = MPAConfig.fromfile(recipe)
         self._patch_datasets(self._recipe_cfg)  # for OTE compatibility
+        self._patch_evaluation(self._recipe_cfg)  # for OTE compatibility
         logger.info(f'initialized recipe = {recipe}')
 
     def _init_model_cfg(self):
         base_dir = os.path.abspath(os.path.dirname(self.template_file_path))
-        return MPAConfig.fromfile(os.path.join(base_dir, 'model.py'))
+        if self._multilabel:
+            cfg_path = os.path.join(base_dir, 'model_multilabel.py')
+        else:
+            cfg_path = os.path.join(base_dir, 'model.py')
+        cfg = MPAConfig.fromfile(cfg_path)
+        cfg.model.multilabel = self._multilabel
```

Comment:
What do you think about declaring this variable as the default value in the base recipe?

Reply:
That makes sense. Let's suppose your suggestion is plan B and current status is plan A. Here is pros and cons about this:
Plan A pros - followed OTE classification convention. Current OTE implementation distingushes multilabel(h-label) or not by the architecture of label schema. Also, another functions are designed to be worked dynamically.
Plan A cons - If we use another classifier, we should pop 'multilabel' argument before build. All the classifiers doesn't take 'multilevel' keyword in their initializer.

Plan B pros - Reverse to Plan A cons, no update is needed in classifier.
Plan B cons - Out of current OTE convention, but I don't think this can be matter.

Plan B (your suggestion) looks better than Plan A (current status). I'll update this.

Reply:
Hi, after review, I realized we can 'only declare' multilabel variable in base recipe and we can't use this as a marker to distinguish whether this task is multi-class or multi-label. Because this the function needs self.multilabel variable to select model.py or model.multilabel. I think determining the kind of task with label schema makes sense.